fppanalysis/two_dim_velocity_estimates.py

```diff
@@ -76,20 +77,113 @@ def _get_dt(ds):
     raise "Unknown format"
 
 
-def _estimate_velocities_given_points(p0, p1, p2, ds):
+def _get_time(x, y, ds):
+    # Sajidah's format
+    if hasattr(ds, "time"):
+        return ds.isel(x=x, y=y).time.values
+
+
+def _estimate_time_delay(
+    x: np.ndarray,
+    x_t: np.ndarray,
+    y: np.ndarray,
+    y_t: np.ndarray,
+    method: str,
+    dt: float,
+    cut_off_freq=1e3,
+    threshold=2.5,
+):
+    """Estimates the average time delay between to signals by finding the time
+    lag that maximizies either the cross-correlation function or the cross
+    conditional average of signal x when signal y is larger than threshold.
+    Either one of these method must be specified in method-argument.
+
+    Input:
+        x: Signal to be conditionally averaged
+        y: Reference signal
+        x_t: Time of signal x
+        y_t: Time of signal y
+        method: 'cross_corr' or 'cond_av' for either cross correlation or cross conditional average
+        if method == 'cross_corr':
+            dt: Time step
+        if method == 'cond_av':
+            cut_off_freq: Cut off frequency to decide window size for running moments
+            threshold: Threshold value for conditional average. Defualt value is set to 2.5.
+
+    Returns:
+        if method == 'cross_corr':
+            td Estimated time delay
+            C Cross correlation at a time lag td.
+
+        if method == 'cond_av':
+            td: Estimated time delay
+            C: Cross conditional variance at a time lag td.
+            events: Number of events larger than 2.5 the mean value
+    """
+
+    if method == "cross_corr":
+        delta_t, c = tde.estimate_time_delay_ccmax(x=x, y=y, dt=dt)
+        return delta_t, c
+
+    if method == "cond_av":
+        delta_t, cond_variance, events = tde.estimate_time_delay_ccond_av_max(
+            x=x, x_t=x_t, y=y, y_t=y_t
+        )
+
+        # Confidence when velocities are estimated from
+        # cross conditional averge is: 1 - conditional variance
+        c = 1 - cond_variance
+
+        return delta_t, c
```

Comment:
You return the same in both if branches, make only one return statement at the end instead.